plugin/pkg/admission/namespace/lifecycle/admission.go

```diff
@@ -82,7 +82,7 @@ func (l *lifecycle) Admit(a admission.Attributes) (err error) {
 		// in case of latency in our caches, make a call direct to storage to verify that it truly exists or not
 		namespaceObj, err = l.client.Namespaces().Get(a.GetNamespace())
 		if err != nil {
-			return admission.NewNotFound(a)
+			return errors.NewNotFound("Namespace", a.GetNamespace())
```

Comment:
only do this if the error from the namespace was a not found error, otherwise return a different error